cql/ManageRareAbnormality.cql

```diff
@@ -627,6 +630,70 @@ define HistologyInterpretedAsCin3:
         aC in Dash."Histologic CIN3"
   )
 
+define AnyHistologyInterpretedAsCin3:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC in Dash."Histologic CIN3"
+  )
+
+define AnyHistologyInterpretedAsCin2:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC ~ Dash."CIN2"
+  )
+
+define MostSevereHistologyInterpretedAsCin2InPast3Years:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC ~ Dash."CIN2"
+  )
+
+define MostRecentCin2BiopsyInPast3Years:
+  Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 3 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."CIN2"
+    )
+  )
+
+define MostRecentCin2BiopsyInPast2Years:
+  Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 2 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."CIN2"
+    )
+  )
+
+define MostRecentHsilBiopsyInPast3Years:
+    Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 3 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."HSIL, Unspecified"
+    )
+  )
+
+ define MostRecentHsilBiopsyInPast2Years:
+    Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 2 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."HSIL, Unspecified"
+    )
+  ) 
```

Comment:
Both of these expressions (`MostRecentHsilBiopsyInPast3Years` and  `MostRecentHsilBiopsyInPast2Years`) should have the name HSIL unspecified in the title rather than just HSIL, since an HSIL biopsy technically refers to a CIN2 or a CIN3, whereas this is an HSIL biopsy that is specifically not specified between the two, and has different recommendations associated with it.

Reply:
Agree - will update the expression name to HsilUnspecified

Reply:
Thanks. There are still a couple of expressions added to CollateManagementData.cql that I believe need to be changed from "Hsil" to "HsilUnspecified". I opened https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/118/files#r1383701330 to address this, and will close this comment since the "Hsil" instances have been resolved within this file.